### ISSUE TITLE
Sequential execution of independent tasks

### ISSUE THREAD

**Issue by jrossyra** [radical-cybertools/radical.pilot#1528]
When continuing with testing for issue #1520 , I also have seen that the tasks are executing sequentially. I submit 2 workloads that each have 5 tasks, plus an additional analysis task in the second workload (which is subject of the other issue). The CUs are running one at a time, so that a task will only begin execution after the completion of a previous one in single file. When I was testing in December this was not the case, so each batch of 5 would execute simultaneously. I'm including all the same execution logs as in #1520 , and the radical stack information.

[rp.agent.tar.gz](https://github.com/radical-cybertools/radical.pilot/files/1635952/rp.agent.tar.gz)
[rp.client.tar.gz](https://github.com/radical-cybertools/radical.pilot/files/1635953/rp.client.tar.gz)

```
(admdrpenv) [jrossyra@titan-ext2 /lustre/atlas/proj-shared/bip149/admdrp/runs/runs/test01]$ radical-stack
2018-01-16 10:37:07,393: radical.saga        : MainProcess                     : MainThread     : INFO    : python.interpreter   version: 2.7.9 (default, May 6 2016, 10:34:30) [GCC 4.9.0 20140422 (Cray Inc.)]
2018-01-16 10:37:07,393: radical.saga        : MainProcess                     : MainThread     : INFO    :                      pid: 12903
2018-01-16 10:37:07,393: radical.saga        : MainProcess                     : MainThread     : INFO    :                      tid: MainThread
2018-01-16 10:37:07,393: radical.saga        : MainProcess                     : MainThread     : INFO    : radical.saga         version: 0.47-v0.46-33-g3c75f23@devel
2018-01-16 10:37:07,436: radical.pilot       : MainProcess                     : MainThread     : INFO    : python.interpreter   version: 2.7.9 (default, May 6 2016, 10:34:30) [GCC 4.9.0 20140422 (Cray Inc.)]
2018-01-16 10:37:07,436: radical.pilot       : MainProcess                     : MainThread     : INFO    :                      pid: 12903
2018-01-16 10:37:07,436: radical.pilot       : MainProcess                     : MainThread     : INFO    :                      tid: MainThread

  python               : 2.7.9
  pythonpath           : /sw/xk6/python/2.7.9/sles11.3_gnu4.3.4/lib/python2.7/site-packages:/lustre/atlas/sw/xk7/python/2.7.9/sles11.3_gnu4.3.4/lib/python2.7/site-packages:/sw/xk6/xalt/0.7.5/site:/sw/xk6/xalt/0.7.5/libexec
  virtualenv           : /lustre/atlas/proj-shared/bip149/jrossyra/admdrpenv

  radical.pilot        : 0.47.1
  radical.utils        : 0.47-v0.47-2-g8b9d4ec@devel
  saga                 : 0.47-v0.46-33-g3c75f23@devel
```

**Comment by andre-merzky** [comment]
The concrrency of the task execution is limited by the concurrency allowed by the backend.  You are running with `resource='ornl.titan_aprun'` I think, and aprun will limit the execution to one task per node.  Thus the sequentiality.

I am not sure about your actual workload, and if aprun is required for some reason - otherwise you may want to switch to `ornl.titan_orte`.

**Comment by jrossyra** [comment]
Ok, I will make this change and see how it goes. Thanks!

**Comment by andre-merzky** [comment]
@jrossyra : did that issue actually resolve with the switch, or are you unable to confirm because of #1538?

**Comment by jrossyra** [comment]
I'm still unable to confirm, but once we resolve #1538 I think that closes this issue and puts me back to testing #1520 if it does not also resolve from changing configs.

**Comment by jrossyra** [comment]
Fixed with use of ORTE branch & resolution of related issues.
